applications/fax/src/fax_maintenance.erl

```diff
@@ -194,6 +194,160 @@ migrate_fax_to_modb(AccountDb, DocId, JObj, Options) ->
+    Headers = props:filter_undefined(
+                [{"Host", kz_json:get_string_value(<<"host">>, FetchRequest)}
+                ,{"Referer", kz_json:get_string_value(<<"referer">>, FetchRequest)}
+                ,{"Content-Type", kz_json:get_string_value(<<"content_type">>, FetchRequest, <<"text/plain">>)}
+                ]),
+    Body = kz_json:get_string_value(<<"content">>, FetchRequest, ""),
+    lager:debug("making ~s request to '~s'", [Method, Url]),
+    case kz_http:req(Method, Url, Headers, Body) of
+        {'ok', 200, RespHeaders, Contents} ->
+            DefaultCt = kz_mime:from_filename(Url),
+            CT = props:get_value("Content-Type", RespHeaders, DefaultCt),
+            ContentType = kz_mime:normalize_content_type(CT),
+            {'ok', Contents, ContentType, 'undefined'};
+        {'ok', Status, _, _} ->
+            ?SUP_LOG_DEBUG("failed to fetch file for job from: ~s, http response: ~b", [Url, Status]);
+        {'error', Reason} ->
+            ?SUP_LOG_DEBUG("failed to fetch file from: ~s for job: ~p", [Url, Reason])
+    end.
+
+-spec update_attachment(kz_json:object(), kz_term:ne_binary(), kz_term:ne_binary(), kz_term:api_binary()) -> 'ok'.
+update_attachment(JObj, Content, ContentType, OldName) ->
+    DocId = kz_doc:id(JObj),
+    lager:debug("converting attachment for doc: ~s", [DocId]),
+    Options = [{<<"output_type">>, 'binary'}
+              ,{<<"job_id">>, DocId}
+              ,{<<"tmp_dir">>, ?TMP_DIR}
+              ,{<<"read_metadata">>, 'true'}
+              ],
+    case kz_convert:fax(ContentType, <<"image/tiff">>, Content, Options) of
+        {'ok', Output, Proplist} ->
+            Values = [{<<"pvt_pages">>, props:get_value(<<"page_count">>, Proplist, 0)}
+                     ,{<<"pvt_size">>, props:get_value(<<"size">>, Proplist, 0)}
+                     ,{<<"pvt_migrated_reason">>, <<"normalize_attachment">>}
+                     ,{<<"pvt_migrated_timestamp">>, kz_time:now_s() }
+                     ],
+            NewDoc = kz_json:set_values(Values, JObj),
+            case fax_util:save_fax_doc(NewDoc, Output, <<"image/tiff">>) of
+                {'ok', NewJObj} ->
+                    lager:debug("updated document saved"),
+                    maybe_delete_old_attachment(NewJObj, OldName);
+                {'error', Error} ->
+                    lager:error("failed saving fax document with message: ~p", [Error])
+            end;
+        {'error', Error} ->
+            lager:error("failed to convert fax document with error: ~p", [Error])
+    end.
+
+-spec maybe_delete_old_attachment(kz_json:object(), kz_term:api_binary()) -> 'ok'.
+maybe_delete_old_attachment(_JObj, 'undefined') ->
+    lager:debug("not deleting attachment, no attachment found");
+maybe_delete_old_attachment(JObj, OldName) ->
+    _ = kz_datamgr:delete_attachment(?KZ_FAXES_DB, kz_doc:id(JObj), OldName),
+    lager:debug("probably deleted attachment from ~s", [OldName]).
```

Comment:
@swysor i think we talked about this. why are we removing the original attachment ?